setup.py

```diff
@@ -57,21 +69,48 @@ def run_tests(self):
         sys.exit(errno)
 
 
+SETUP_COMMANDS['test'] = PyTestCommand
+
+
 class BuildDocsCommand(setuptools.command.build_py.build_py):
-    apidoc_command = (
-        'sphinx-apidoc', '-f', '-o', 'docs',
-        '--no-toc',
-        'coalib'
-    )
-    make_command = ('make', '-C', 'docs', 'html', 'SPHINXOPTS=-W')
+
+    def initialize_options(self):
+        setup_dir = os.path.join(os.getcwd(), __dir__)
+        docs_dir = os.path.join(setup_dir, 'docs')
+        source_docs_dir = os.path.join(setup_dir, 'docs')
+
+        set_python_path(setup_dir)
+
+        self.apidoc_commands = list()
+
+        self.apidoc_commands.append((
+            'sphinx-apidoc', '-f', '-o', source_docs_dir,
+            '--no-toc',
+            os.path.join(setup_dir, 'coalib')
+        ))
```

Comment:
Why double braces?

Reply:
Got it 😄

Reply:
the inner braces are a `tuple`.

Reply:
also, this is an imported template.  CI already verifies that it doesnt break somethings, but `setup.py` isnt extensively tested in CI.

If there is a problem with the template , raise a bug in the source repo.